scripts/hdl/simify_verilog.py

```diff
@@ -31,6 +31,28 @@ def add_ispm(contents: str) -> str:
 
     return contents.replace(orig_str, new_str)
 
+def add_cycle(contents: str) -> str:
+    """
+    Print cycle-by-cycle statements
+    """
+    m = re.search(r'module Datapath\([\s\S]+?\);', contents, re.MULTILINE)
+    assert m is not None
+
+    orig_str = m.group()
+    new_str = f"""
+
+always @(posedge clock) begin
+
+  if (1'($test$plusargs("cycle"))) begin
+    // the default tid is 0
+    $display("[0] pc = [0x%h] inst = [0x%h] DASM(0x%h)", exe_reg_pc, dec_reg_inst, dec_reg_inst);
+  end
```

Comment:
Is it not possible to print this from Chisel?

Reply:
It is possible to print from Chisel, but we had issues passing a command-line argument into Chisel. The plus arguments of Verilog seemed to be a straightforward way to do this, so we implemented it this way. What do you think?